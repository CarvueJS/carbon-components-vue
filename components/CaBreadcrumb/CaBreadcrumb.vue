<template>
  <nav class="bx--breadcrumb bx--breadcrumb--no-trailing-slash" aria-label="breadcrumb">
    <slot></slot>
  </nav>
</template>

<script>
  export default {
    name: 'ca-breadcrumb',
  };
</script>

<style></style>
