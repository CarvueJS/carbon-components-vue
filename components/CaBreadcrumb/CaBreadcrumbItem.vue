<template>
  <div class="bx--breadcrumb-item">
    <a :href="href" class="bx--link">
      <slot></slot>
    </a>
  </div>
</template>

<script>
  export default {
    name: 'ca-breadcrumb-item',
    props: {
      href: {
        type: String,
        default: '#'
      }
    }
  };
</script>

<style></style>
