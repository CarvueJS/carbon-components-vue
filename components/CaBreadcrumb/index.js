import CaBreadcrumb from './CaBreadcrumb.vue';
import CaBreadcrumbItem from './CaBreadcrumbItem.vue';

CaBreadcrumb.install = Vue => {
  Vue.component(CaBreadcrumb.name, CaBreadcrumb);
  Vue.component(CaBreadcrumbItem.name, CaBreadcrumbItem);
}

export default CaBreadcrumb
