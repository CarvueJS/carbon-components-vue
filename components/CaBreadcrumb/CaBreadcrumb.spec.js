import { shallow } from 'vue-test-utils';
import CaBreadcrumb from './CaBreadcrumb.vue';
import CaBreadcrumbItem from './CaBreadcrumbItem.vue';

describe('CaBreadcrumb', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaBreadcrumb, { 
      slots: {
        default: CaBreadcrumbItem
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot: without last slash', () => {
      const wrapper = shallow(CaBreadcrumb, { 
        propsData: {
          noSlash: true
        },
        slots: {
          default: [CaBreadcrumbItem, CaBreadcrumbItem]
        }
      }
    )
    expect(wrapper.html()).toMatchSnapshot()
  });

  test('Mathes snapshot: with last slash', () => {
    const wrapper = shallow(CaBreadcrumb, { 
      slots: {
        default: [CaBreadcrumbItem, CaBreadcrumbItem]
      }
    }
  )
  expect(wrapper.html()).toMatchSnapshot()
});
})
