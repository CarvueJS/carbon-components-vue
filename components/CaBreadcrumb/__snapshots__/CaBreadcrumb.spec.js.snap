// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaBreadcrumb Mathes snapshot: with last slash 1`] = `
<nav aria-label="breadcrumb" class="bx--breadcrumb">
  <div class="bx--breadcrumb-item">
    <a href="#" class="bx--link"></a>
  </div>
  <div class="bx--breadcrumb-item">
    <a href="#" class="bx--link"></a>
  </div>
</nav>
`;

exports[`CaBreadcrumb Mathes snapshot: without last slash 1`] = `
<nav aria-label="breadcrumb" class="bx--breadcrumb bx--breadcrumb--no-trailing-slash">
  <div class="bx--breadcrumb-item">
    <a href="#" class="bx--link"></a>
  </div>
  <div class="bx--breadcrumb-item">
    <a href="#" class="bx--link"></a>
  </div>
</nav>
`;
